flink-python/src/main/java/org/apache/flink/streaming/api/runners/python/beam/state/BeamStateHandler.java

```diff
@@ -23,6 +23,35 @@
 /** Interface for doing actual operations on Flink state based on {@link BeamFnApi.StateRequest}. */
 public interface BeamStateHandler<S> {
 
-    BeamFnApi.StateResponse.Builder handle(BeamFnApi.StateRequest request, S state)
+    /**
+     * Dispatches {@link BeamFnApi.StateRequest} to different handle functions base on request case.
+     */
+    default BeamFnApi.StateResponse.Builder handle(BeamFnApi.StateRequest request, S state)
```

Comment:
I wonder if it would be clearer if we introduce an abstract class of `AbstractBeamStateHandler`. Personally, I don't like the form of default interface methods.

Reply:
That's a good point.